src/serde_utils.rs

```diff
@@ -50,7 +50,7 @@ impl<'de, F, T, R> Visitor<'de> for MappedSequenceVisitor<T, R, F>
     fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
         where A: SeqAccess<'de>,
     {
-        let mut v = Vec::new();
+        let mut v = Vec::with_capacity(seq.size_hint().unwrap_or_else(|| 0));
```

Comment:
This one actually requires the most research. Is the sequence size hint untrusted user input? How do deserializers usually handle this. We only want to oom if the serialization input has size to match.

Reply:
Ok, serde itself uses a cautious approach when deserializing to `Vec`. It seems like the right thing to do here too. In serde we can find they use this:

```rust
    #[inline]
    pub fn cautious(hint: Option<usize>) -> usize {
        cmp::min(hint.unwrap_or(0), 4096)
    }
```